Automattic/jetpack
# Search: Fix search pricing request before site is connected

#### Changes proposed in this Pull Request:
The PR fixes a regression introduced by #24795. `get_pricing_from_wpcom` would be called before site is connected, meaning we wouldn't be able to `wpcom_json_api_request_as_blog`. The PR also added `my-jetpack` as dependency for the `search` package.

#### Jetpack product discussion
n/a

#### Does this pull request change what data or activity we track or use?
no

#### Testing instructions:
* Go to a fresh created JN site (or ensure blog is not connected, blog token is empty and the blog doesn't have search subscription)
* Ensure search pricing is correct on page `/wp-admin/admin.php?page=jetpack-search`
* Connect the site
* Ensure search pricing is correct in My Jetpack `/wp-admin/admin.php?page=my-jetpack#/add-search`
